Objects/object.c

```diff
@@ -282,13 +282,10 @@ PyObject_Print(PyObject *op, FILE *fp, int flags)
                 s = PyObject_Str(op);
             else
                 s = PyObject_Repr(op);
-            if (s == NULL)
+            if (s == NULL) {
                 ret = -1;
-            else if (PyBytes_Check(s)) {
-                fwrite(PyBytes_AS_STRING(s), 1,
-                       PyBytes_GET_SIZE(s), fp);
             }
-            else if (PyUnicode_Check(s)) {
+            else {
                 PyObject *t;
                 t = PyUnicode_AsEncodedString(s, "utf-8", "backslashreplace");
```

Comment:
While you are cleaning up: creating a bytes object here is extra work. Python strings already have a UTF-8 representation available. You can replace everything that deals with `t` by using `PyUnicode_AsUTF8AndSize`.
(FWIW, `errors="backslashreplace"` does nothing: `utf-8` can encode any string.)

Reply:
Done it. Please have a look to see if it is correct. I've ran the tests and they were successful.